File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/meta/store/RendezvousStoreAddResponse.scala
package com.webank.eggroll.core.deepspeed.meta.store

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class RendezvousStoreAddResponse(amount: Long)

object RendezvousStoreAddResponse {
  implicit def serialize(src: RendezvousStoreAddResponse): Array[Byte] = {
    val builder = Deepspeed.StoreAddResponse.newBuilder()
      .setAmount(src.amount)
    builder.build().toByteArray
  }

  implicit def deserialize(byteString: ByteString): RendezvousStoreAddResponse = {
    val src = Deepspeed.StoreAddResponse.parseFrom(byteString)
    RendezvousStoreAddResponse(
      amount = src.getAmount
    )
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/deepspeed/store/RendezvousStoreAddResponse.java
package com.eggroll.core.deepspeed.store;

import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Deepspeed;

public class RendezvousStoreAddResponse {
    private long amount;

    public RendezvousStoreAddResponse(long amount) {
        this.amount = amount;
    }

    public long getAmount() {
        return amount;
    }

    public void setAmount(long amount) {
        this.amount = amount;
    }

    public static byte[] serialize(RendezvousStoreAddResponse src) {
        Deepspeed.StoreAddResponse.Builder builder = Deepspeed.StoreAddResponse.newBuilder()
                .setAmount(src.getAmount());
        return builder.build().toByteArray();
    }

    public static RendezvousStoreAddResponse deserialize(byte[] bytes) throws InvalidProtocolBufferException {
        Deepspeed.StoreAddResponse src = Deepspeed.StoreAddResponse.parseFrom(bytes);
        return new RendezvousStoreAddResponse(
                src.getAmount()
        );
    }
}